tests/functional/test_k8s_common.py

```diff
@@ -80,3 +80,9 @@ def test_atomic_updates(self):
         assert cfg_data_1 != cfg_data_2
         assert "old_password" in cfg_data_1
         assert "new_password" in cfg_data_2
+
+    @mock.patch("charmhelpers.core.hookenv.network_get", autospec=True)
+    def test_get_ingress_address(self, network_get):
+        network_get.return_value = {"ingress-addresses": ["1.2.3.4", "5.6.7.8"]}
+        ingress = kubernetes_common.get_ingress_address("endpoint-name", ["1.2.3.4"])
```

Comment:
We should test the existing calling convention as well to ensure no regressions.

```suggestion
        ingress = kubernetes_common.get_ingress_address("endpoint-name")
        assert ingress == "1.2.3.4"
        ingress = kubernetes_common.get_ingress_address("endpoint-name", ["1.2.3.4"])
```

Reply:
good catch. new version of the patch captures this.